internal/server/server.go

```diff
@@ -193,15 +193,17 @@ func (h *Handler) Predict(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if h.cfg.UseProcedureMode {
-		if req.ProcedureSourceURL == "" || req.Token == "" {
+		procedureSourceUrl := req.Context["procedure_source_url"].(string)
+		token := req.Context["token"].(string)
```

Comment:
We need to check if `"procedure_source_url"` & `"token"` are in `Context` first, or these 2 `(.string)` casts will NPE?